File: MauiNenuExamGrades/GradeRecord.cs
namespace MauiNenuExamGrades
{
    public class GradeRecord : IComparable
    {
        private string _TotalScore;
        private List<string> _UsualScore;

        public string TotalScore { get { return _TotalScore; } set { _TotalScore = value; } }
        public List<string> UsualScore { get { return _UsualScore; } set { _UsualScore = value; } }

        public GradeRecord() { }

        public GradeRecord(string totalRecord) { _TotalScore = totalRecord; }

        public int CompareTo(object obj)
        {
            GradeRecord other = obj as GradeRecord;
            int res1, res2;
            if (other != null)
            {
                if (!int.TryParse(this._TotalScore, out res1))
                    res1 = 0;
                if (!int.TryParse(other._TotalScore, out res2))
                    res2 = 0;
                if (res1 < res2)
                    return -1;
                else if (res2 < res1)
                    return 1;
                else
                    return 0;
            }
            else
            {
                return 1;
            }
        }

        public override string ToString()
        {
            return _TotalScore;
        }
    }
}

File: NenuExamGrades/GradeRecord.vb
Public Class GradeRecord
    Implements IComparable
    Private __totalScore As String
    Private __usualScore As List(Of String)

    Property TotalScore As String
        Get
            Return __totalScore
        End Get
        Set(value As String)
            __totalScore = value
        End Set
    End Property

    Property UsualScore As List(Of String)
        Get
            Return __usualScore
        End Get
        Set(value As List(Of String))
            __usualScore = value
        End Set
    End Property

    Sub New(_totalScore As String)
        __totalScore = _totalScore
    End Sub

    Public Function CompareTo(obj As Object) As Integer Implements IComparable.CompareTo
        Dim gr2 As GradeRecord = obj
        Dim res1, res2 As Integer
        If Not Integer.TryParse(Me.TotalScore, res1) Then
            res1 = 0
        End If
        If Not Integer.TryParse(gr2.TotalScore, res2) Then
            res2 = 0
        End If
        If res1 < res2 Then
            Return -1
        ElseIf res1 > res2 Then
            Return 1
        Else
            Return 0
        End If
    End Function
End Class
